tests/integration/pkg/helpers/api_rule.go

```diff
@@ -31,19 +31,36 @@ type apiRuleStatusV2Alpha1 struct {
 // RetryableApiRule wraps any function that modifies or creates an APIRule
 type RetryableApiRule func(k8sClient dynamic.Interface, resources ...unstructured.Unstructured) (*unstructured.Unstructured, error)
 
-// APIRuleWithRetries tries toExecute function and retries with onRetry if APIRule status is "ERROR"
+// APIRuleWithRetries tries toExecute function and retries with onRetry if APIRule status is "ERROR" or "Error"
 func ApplyApiRule(toExecute RetryableApiRule, onRetry RetryableApiRule, k8sClient dynamic.Interface, retryOpts []retry.Option, resources []unstructured.Unstructured) error {
 	res, err := toExecute(k8sClient, resources...)
 	if err != nil {
 		return err
 	}
 
-	apiStatus, err := GetAPIRuleStatus(res)
-	if err != nil {
-		return err
+	var arVersion, code, description string
+
+	if res.Object != nil && res.Object["apiVersion"] != nil {
+		apiVersion := strings.Split(res.Object["apiVersion"].(string), "/")
+
+		if apiVersion[1] == "v1beta1" {
+			arVersion = apiVersion[1]
+			arStatus, err := GetAPIRuleStatus(res)
+			if err != nil {
+				return err
+			}
+			code = arStatus.Status.APIRuleStatus.Code
+		} else if apiVersion[1] == "v2alpha1" {
+			arVersion = apiVersion[1]
+			arStatus, err := GetAPIRuleStatusV2Alpha1(res)
+			if err != nil {
+				return err
+			}
+			code = arStatus.Status.State
+		}
 	}
 
-	if apiStatus.Status.APIRuleStatus.Code == "ERROR" {
+	if code == "ERROR" || code == "Error" {
```

Comment:
I do not want to refactor the code / change behaviour. Just fix the bug.

Reply:
The problem I see is if the check for empty code is not fixed, the whole checking for different error code is meaningless because it is never relevant. Therefore we need to consider it.

Reply:
Yes, and it was like that before. Logic of the code is only if status is ERROR. I do not like that logic, it should basically check if APIRule is not Ready `!= "Ready"` like in [ApplyApiRuleV2Alpha1](https://github.com/kyma-project/api-gateway/blob/569011738360f5c624d251965b33b14b6bb37e2e/tests/integration/pkg/helpers/api_rule.go#L71).
As I noted before, I do not want to refactor or change existing logic. This could be part of separate task.

Reply:
If we see code that is not working as intended, we should fix it. Adding "Error" check and additional status reading is just adding dead code, because it will never be executed since status code is empty and therefore not even one execution of the code will happen.